app/containers/sync/index.js

```diff
@@ -0,0 +1,69 @@
+import { connect } from 'react-redux';
+import { getAreas } from 'redux-modules/areas';
+import { getFeedbackQuestions } from 'redux-modules/feedback';
+import { getReportQuestions } from 'redux-modules/reports';
+import { getCountries } from 'redux-modules/countries';
+import { getLanguage } from 'helpers/language';
+import { getReadyState } from 'helpers/sync';
+import { getUser } from 'redux-modules/user';
+
+import Sync from 'components/sync';
+
+const getForms = ({ hasForms, languageChanged, getFormsDispatch }) => {
+  if (!hasForms.report || languageChanged) getFormsDispatch.report();
+  if (!hasForms.daily || languageChanged) getFormsDispatch.daily();
+  if (!hasForms.weekly || languageChanged) getFormsDispatch.weekly();
+};
+
+function mapStateToProps(state) {
+  return {
+    isConnected: state.offline.online,
+    reach: state.app.netInfo.reach,
+    languageChanged: state.app.language !== getLanguage(),
+    hasAreas: state.areas.data && state.areas.data.length > 0 && state.areas.synced,
+    hasUserData: state.user.data && Object.keys(state.user.data).length > 0 && state.user.synced,
+    hasForms: {
+      report: Object.keys(state.reports.forms).length > 0 && state.reports.synced,
+      daily: Object.keys(state.feedback.daily).length > 0 && state.feedback.dailySynced,
+      weekly: Object.keys(state.feedback.weekly).length > 0 && state.feedback.weeklySynced
+    },
+    readyState: getReadyState(state)
+  };
+}
+
+function mapDispatchToProps(dispatch) {
+  return {
+    getAreasDispatch: () => {
+      dispatch(getAreas());
+    },
+    getUserDispatch: () => {
+      dispatch(getUser());
+    },
+    getCountries: () => {
+      dispatch(getCountries());
+    },
+    getFormsDispatch: {
+      report: () => dispatch(getReportQuestions()),
+      daily: () => dispatch(getFeedbackQuestions('daily')),
+      weekly: () => dispatch(getFeedbackQuestions('weekly'))
+    }
+  };
+}
+
+function mergeProps({ hasUserData, hasAreas, hasForms, languageChanged, ...stateProps },
+  { getUserDispatch, getAreasDispatch, getFormsDispatch, ...dispatchProps }, ownProps) {
+  return {
+    ...ownProps,
+    ...dispatchProps,
+    ...stateProps,
+    getUser: () => (!hasUserData && getUserDispatch()),
```

Comment:
Do you like the idea to move this to a new action so we will check it there instead to check all of the flags in the component? Also if we move the logic to an action we will be able to sync the app from any place.